Formula/gcc.rb

```diff
@@ -120,16 +120,42 @@ def install
       inreplace "libgcc/config/t-slibgcc-darwin", "@shlib_slibdir@", "#{HOMEBREW_PREFIX}/lib/gcc/#{version_suffix}"
     end
 
+    on_linux do
+      # Fix cc1: error while loading shared libraries: libisl.so.15
+      args << "--with-boot-ldflags=-static-libstdc++ -static-libgcc #{ENV["LDFLAGS"]}"
+
+      # Fix Linux error: gnu/stubs-32.h: No such file or directory.
+      args << "--disable-multilib"
+
+      # Change the default directory name for 64-bit libraries to `lib`
+      # https://stackoverflow.com/a/54038769
+      inreplace "gcc/config/i386/t-linux64", "m64=../lib64", "m64="
+    end
+
     mkdir "build" do
       system "../configure", *args
 
-      # Use -headerpad_max_install_names in the build,
-      # otherwise updated load commands won't fit in the Mach-O header.
-      # This is needed because `gcc` avoids the superenv shim.
-      system "make", "BOOT_LDFLAGS=-Wl,-headerpad_max_install_names"
-      system "make", "install"
+      on_macos do
+        # Use -headerpad_max_install_names in the build,
+        # otherwise updated load commands won't fit in the Mach-O header.
+        # This is needed because `gcc` avoids the superenv shim.
+        system "make", "BOOT_LDFLAGS=-Wl,-headerpad_max_install_names"
+        system "make", "install"
+      end
+
+      on_linux do
+        system "make"
+        system "make", "install-strip"
+      end
 
       bin.install_symlink bin/"gfortran-#{version_suffix}" => "gfortran"
+
+      on_linux do
+        # Only the newest brewed gcc should install gfortan libs as we can only have one.
+        lib.install_symlink lib/"gcc/#{version_suffix}/libgfortran.so"
+        lib.install_symlink lib/"gcc/#{version_suffix}/libgfortran.a"
+        lib.install_symlink lib/"gcc/#{version_suffix}/libgfortran.so.5"
+      end
```

Comment:
Is there any reason we don't do this on macOS too?

Reply:
Also, any reason not to do this:
```suggestion
      on_linux do
        # Only the newest brewed gcc should install gfortan libs as we can only have one.
        lib.install_symlink Dir[lib/"gcc/#{version_suffix}/libgfortran.*"]
      end
```

Reply:
I changed the syntax to be more generic as asked above.

To the "why do you need to do this on Linux": if I don't the linker will fail to find the libraries as they are in the `gcc/10.2.0_4/lib/gcc/10` subfolder. But I don't really know why this is not needed on macOS. Maybe someone from @Homebrew/linux  might have an idea.